### PR TITLE
Refactor `add_weighted_loan_emission_factor`

### LINKED ISSUE #137: All functions that require `loan_size_*` should check if input data has identical currency
All "loan-weighting" functions (ie. `summarize_*`) assume that the input currency of `loanbook` is consistent (identical across all loans). It doesn't matter what the actual currency is, so long as it's the same across all loans. 

We should write a check for this at the beginning of all functions that have `loan_size_*` as a crucial column, and the functions should error if currencies are not identical. 

Thanks @QianFeng2020

### LINKED ISSUE #177: Extract loan-weighting function of `emission_factor` from `target_sda()` to summarize_weighted_production.R
Internal: `summarize_weighted_production()` has two `loan-weighting`- type functions (for `production` and `percent_change`) while a separate function exists to weight `emission_factor` inside `target_sda`. 

To be DRY, I think we should extract this, and try to keep all functions related to loan-weighting in summarize_weighted_production.R

Relates to #131, #137 and #165

### PR DESCRIPTION
**Pull request by jdhoffa**
Closes #177 and closes #137

### CODE REVIEW

**Comment by maurolepore** [comment]
Is this ready for review?

**Comment by jdhoffa** [comment]
It is yes!

File: R/summarize_weighted_production.R [hunk 1]
```diff
@@ -86,40 +86,56 @@ summarize_weighted_metric <- function(data,
...
-  add_weighted_loan_metric(data, use_credit_limit, percent_change = FALSE)
+  add_weighted_loan_metric(data, use_credit_limit, metric = "production")
 }
 
-add_weighted_loan_metric <- function(data, use_credit_limit, percent_change) {
+add_weighted_loan_emission_factor <- function(data, use_credit_limit = FALSE) {
+  add_weighted_loan_metric(data, use_credit_limit, metric = "emission_factor")
+}
+
+add_weighted_loan_metric <- function(data, use_credit_limit, metric) {
```
Comment (maurolepore) [inline R/summarize_weighted_production.R:100]: Consider enumerating options to `metric` 
-- https://principles.tidyverse.org/def-enum.html

File: tests/testthat/test-summarize_weighted_production.R [hunk 2]
```diff
@@ -420,3 +420,24 @@ test_that("with known input outputs as expected", {
     summarize_weighted_percent_change(name_ald, use_credit_limit = TRUE)
   expect_equal(out2$weighted_percent_change, c(0, 0, 40, 180))
 })
+
+test_that("with duplicated currencies errors with informative message (#137)", {
```
Comment (maurolepore) [inline tests/testthat/test-summarize_weighted_production.R:424]: Is "duplicated" the best word? The error class suggests "different currencies" is better than "duplicated currencies".

File: R/summarize_weighted_production.R [hunk 3]
```diff
@@ -86,40 +86,56 @@ summarize_weighted_metric <- function(data,
...
   stopifnot(
     is.data.frame(data),
-    isTRUE(use_credit_limit) || isFALSE(use_credit_limit)
+    isTRUE(use_credit_limit) || isFALSE(use_credit_limit),
+    metric %in% c('production', "percent_change", "emission_factor")
```
Comment (maurolepore) [inline R/summarize_weighted_production.R:104]: If we use `match.arg()` we man not need this.

File: R/summarize_weighted_production.R [hunk 4]
```diff
@@ -86,40 +86,56 @@ summarize_weighted_metric <- function(data,
...
     "production",
     "sector_ald",
-    "technology",
     "year"
   )
 
-  if (percent_change) {
+  check_identical_currencies(data, currency)
+
+  if (metric %in% c("production", "percent_change")) {
```
Comment (maurolepore) [inline R/summarize_weighted_production.R:127]: Maybe `switch()` could help write a clearer implementation -- I'll try.

File: tests/testthat/test-target_sda.R [hunk 5]
```diff
@@ -206,49 +206,49 @@ test_that("with known input outputs as expected", {
...
-  expect_equal(out_shaanxi$target_sds$emission_factor_value, c(0.9, 5.16))
-
-  out_company_2 <- filter(out, name_ald == "company 2") %>%
-    split(.$emission_factor_metric)
-
-  expect_equal(out_company_2$projected$emission_factor_value, 12)
-  expect_equal(out_company_2$target_b2ds$emission_factor_value, 12)
-  expect_equal(out_company_2$target_sds$emission_factor_value, 12)
-})
+# test_that("with known input outputs as expected, at company level (#155)", {
```
Comment (maurolepore) [inline tests/testthat/test-target_sda.R:209]: Is this dead code?

**Comment by maurolepore** [comment]
@jdhoffa, thanks -- looks good. Please address my only comment above then merge whenever you are ready.

File: tests/testthat/test-summarize_weighted_production.R (hunk 2 above)
Comment (jdhoffa) [reply tests/testthat/test-summarize_weighted_production.R]: yes you are right

**Pull request merged**